simulators/ethereum/pyspec/types.go

```diff
@@ -0,0 +1,181 @@
+	BestBlock  common.UnprefixedHash `json:"lastblockhash"`
+	Network    string                `json:"network"`
+	SealEngine string                `json:"sealEngine"`
+}
+
+type block struct {
+	Rlp          string         `json:"rlp"`
+	BlockHeader  *blockHeader   `json:"blockHeader"`
+	Transactions []transactions `json:"transactions"`
+	UncleHeaders []byte         `json:"uncleHeaders"`
+	Withdrawals  []withdrawals  `json:"withdrawals"`
+	Exception    string         `json:"expectException"`
+}
+
+//go:generate go run github.com/fjl/gencodec -type blockHeader -field-override blockHeaderUnmarshaling -out gen_bh.go
+type blockHeader struct {
+	ParentHash          common.Hash      `json:"parentHash"`
+	UncleHash           common.Hash      `json:"sha3Uncles"`
+	UncleHashAlt        common.Hash      `json:"uncleHash"` // name in fixtures
+	Coinbase            common.Address   `json:"coinbase"`
+	CoinbaseAlt         common.Address   `json:"author"` // nethermind/parity/oe name
+	CoinbaseAlt2        common.Address   `json:"miner"`  // geth/besu name
+	StateRoot           common.Hash      `json:"stateRoot"`
+	TransactionsTrie    common.Hash      `json:"transactionsRoot"`
+	TransactionsTrieAlt common.Hash      `json:"transactionsTrie"` // name in fixtures
+	ReceiptTrie         common.Hash      `json:"receiptsRoot"`
+	ReceiptTrieAlt      common.Hash      `json:"receiptTrie"` // name in fixturse
+	Bloom               types.Bloom      `json:"bloom"`
+	Difficulty          *big.Int         `json:"difficulty"`
+	Number              *big.Int         `json:"number"`
+	GasLimit            uint64           `json:"gasLimit"`
+	GasUsed             uint64           `json:"gasUsed"`
+	Timestamp           *big.Int         `json:"timestamp"`
+	ExtraData           []byte           `json:"extraData"`
+	MixHash             common.Hash      `json:"mixHash"`
+	Nonce               types.BlockNonce `json:"nonce"`
+	BaseFee             *big.Int         `json:"baseFeePerGas"`
+	Hash                common.Hash      `json:"hash"`
+	WithdrawalsRoot     common.Hash      `json:"withdrawalsRoot"`
+}
+
+type blockHeaderUnmarshaling struct {
+	Difficulty *math.HexOrDecimal256 `json:"difficulty"`
+	Number     *math.HexOrDecimal256 `json:"number"`
+	GasLimit   math.HexOrDecimal64   `json:"gasLimit"`
+	GasUsed    math.HexOrDecimal64   `json:"gasUsed"`
+	Timestamp  *math.HexOrDecimal256 `json:"timestamp"`
+	ExtraData  hexutil.Bytes         `json:"extraData"`
+	BaseFee    *math.HexOrDecimal256 `json:"baseFeePerGas"`
+}
+
+//go:generate go run github.com/fjl/gencodec -type transactions -field-override transactionsUnmarshaling -out gen_txs.go
+type transactions struct {
+	Nonce     uint64         `json:"nonce"`
+	To        common.Address `json:"to.omitempty"`
+	Value     *big.Int       `json:"value"`
+	Data      []byte         `json:"data"`
+	GasLimit  uint64         `json:"gasLimit"`
+	GasUsed   uint64         `json:"gasUsed"`
+	SecretKey common.Hash    `json:"secretKey"`
+}
```

Comment:
There are some missing fields that should be included from here:
https://github.com/ethereum/execution-spec-tests/blob/2dcfb503f4c50d368958c34a99b39e1aa3f62238/src/ethereum_test_tools/common/types.py#L912-L940